### PR TITLE
Remove handling of events from `stream-to-file`.

### LINKED ISSUE #113: Trace saving solution
The existing [`stream-to-file`](https://github.com/STFC-ICD-Research-and-Design/supermusr-data-pipeline/tree/main/stream-to-file/src) tool should be adapted to handle saving traces for development/diagnostics.

- Remove handling of events, this is handled by the dedicated NeXus writer
- Allow the ability to consume a control topic, this will handle receiving run start and run stop messages
- If no control topic is provided then the tool behaves as it does now, i.e. continuously save data until it is terminated
- If a control topic is provided then the tool outputs a new HDF file for each run (i.e. from when a run start message is received, up until a run stop message is received)

### PR DESCRIPTION
**Pull request by Henry-Denny**
**Overview of Changes:**

- `stream-to-file` has been renamed to `trace-archiver-hdf5`.
- The handling of events functionality has been removed from this tool.

This partially resolves #113.

### CODE REVIEW

File: docs/july_test.md [hunk 1]
```diff
@@ -4,9 +4,7 @@
```
Comment (DanNixon) [inline docs/july_test.md:4]: This is a record of what was done, it should be left as it is.

File: trace-archiver-hdf5/src/main.rs [hunk 2]
```diff
@@ -99,22 +77,13 @@ async fn main() -> Result<()> {
     .set("enable.auto.commit", "false")
     .create()?;
 
-    let topics_to_subscribe: Vec<String> = vec![args.event_topic, args.trace_topic]
-        .into_iter()
-        .flatten()
-        .collect();
-    let topics_to_subscribe: Vec<&str> = topics_to_subscribe.iter().map(|i| i.as_ref()).collect();
-    if topics_to_subscribe.is_empty() {
+    if let Some(trace_topic) = args.trace_topic {
```
Comment (DanNixon) [inline trace-archiver-hdf5/src/main.rs:80]: Trace topic should be made to be required now.

**Review by DanNixon — CHANGES_REQUESTED** [review]

File: trace-archiver-hdf5/src/main.rs (hunk 2 above)
Comment (Henry-Denny) [reply trace-archiver-hdf5/src/main.rs:80]: Should the trace file also be required?

Comment (DanNixon) [reply trace-archiver-hdf5/src/main.rs:80]: Yes.

File: trace-archiver-hdf5/src/main.rs [hunk 3]
```diff
@@ -31,10 +31,10 @@ struct Cli {
...
     #[clap(long)]
-    trace_topic: Option<String>,
+    trace_topic: String,
 
     #[clap(long)]
-    trace_file: Option<PathBuf>,
+    trace_file: PathBuf,
 
     #[clap(long)]
     digitizer_count: Option<usize>,
```
Comment (DanNixon) [inline trace-archiver-hdf5/src/main.rs:40]: This can now be made to be required.

**Review by DanNixon — CHANGES_REQUESTED** [review]

File: trace-archiver-hdf5/src/main.rs [hunk 4]
```diff
@@ -38,19 +31,13 @@ struct Cli {
     consumer_group: String,
 
     #[clap(long)]
-    event_topic: Option<String>,
+    trace_topic: String,
```
Comment (DanNixon) [inline trace-archiver-hdf5/src/main.rs:34]:
```suggestion
    topic: String,
```

File: trace-archiver-hdf5/src/main.rs [hunk 5]
```diff
@@ -38,19 +31,13 @@ struct Cli {
...
 
     #[clap(long)]
-    event_file: Option<PathBuf>,
+    trace_file: PathBuf,
```
Comment (DanNixon) [inline trace-archiver-hdf5/src/main.rs:37]:
```suggestion
    file: PathBuf,
```

**Review by DanNixon — CHANGES_REQUESTED** [review]

File: trace-archiver-hdf5/src/main.rs (hunk 4 above)
Comment (Henry-Denny) [reply trace-archiver-hdf5/src/main.rs]: Will this not be confusing once there is also a `control_topic` command line argument?

Comment (DanNixon) [reply trace-archiver-hdf5/src/main.rs]: Ah, yes. In that case keep this as `trace_topic`.

**Review by DanNixon — APPROVED** [review]

**Pull request merged**